blob/rollup.config.js

```diff
@@ -0,0 +1,24 @@
+import localResolve from 'rollup-plugin-local-resolve';
+import nodeResolve from 'rollup-plugin-node-resolve';
+import { uglify } from 'rollup-plugin-uglify';
+
+export default [
+  {
+    external: ['ms-rest-js', 'crypto', 'fs', 'events', 'os'],
+    input: 'dist-esm/lib/index.js',
```

Comment:
Seems here are using `dist-esm/lib/index.js` as the source of rollup to generate `dist/index.js`. But things are more complex. There are 2 index.ts defined, `index.ts` for Node.js and `index.browsers.ts` for browsers. Because we want to hide some interfaces in browsers, such as sharedkey based auth.

So we need 2+ rollup builds, one for Node.js, based on `dist-esm/lib/index.js` as the entry for Node.js
Another build is for browsers, based on dist-esm/lib/index.browsers.js` as the entry for browsers bundle.
For browser testing, the 3rd rollup building is needed to generate a test bundle.

During generating the browser builds, currently "events" is required to be added as a browser shim. `os` `crypto` and `fs` are not required.

Reply:
I tried the rollup build up, the Node.js build works great. This PR is excellent!